packages/expo/cli/start/project/ProjectDevices.ts

```diff
@@ -0,0 +1,96 @@
+import { createTemporaryProjectFile } from './dotExpo';
+
+export type DeviceInfo = {
+  installationId: string;
+  lastUsed: number;
+};
+
+export type DevicesInfo = {
+  devices: DeviceInfo[];
+};
+
+const DEVICES_FILE_NAME = 'devices.json';
+
+const MILLISECONDS_IN_30_DAYS = 30 * 24 * 60 * 60 * 1000;
+
+export const DevicesFile = createTemporaryProjectFile<DevicesInfo>(DEVICES_FILE_NAME, {
+  devices: [],
+});
+
+let devicesInfo: DevicesInfo | null = null;
+
+export async function getDevicesInfoAsync(projectRoot: string): Promise<DevicesInfo> {
+  if (devicesInfo) {
+    return devicesInfo;
+  }
+  return readDevicesInfoAsync(projectRoot);
+}
+
+export async function readDevicesInfoAsync(projectRoot: string): Promise<DevicesInfo> {
+  try {
+    devicesInfo = await DevicesFile.readAsync(projectRoot);
+
+    // if the file on disk has old devices, filter them out here before we use them
+    const filteredDevices = filterOldDevices(devicesInfo.devices);
+    if (filteredDevices.length < devicesInfo.devices.length) {
+      devicesInfo = {
+        ...devicesInfo,
+        devices: filteredDevices,
+      };
+      // save the newly filtered list for consistency
+      try {
+        await setDevicesInfoAsync(projectRoot, devicesInfo);
+      } catch {
+        // do nothing here, we'll just keep using the filtered list in memory for now
+      }
+    }
+
+    return devicesInfo;
+  } catch {
+    return await DevicesFile.setAsync(origin, { devices: [] });
+  }
+}
+
+export async function setDevicesInfoAsync(
+  projectRoot: string,
+  json: DevicesInfo
+): Promise<DevicesInfo> {
+  devicesInfo = json;
+  return await DevicesFile.setAsync(projectRoot, json);
+}
+
+export async function saveDevicesAsync(
+  projectRoot: string,
+  deviceIds: string | string[]
+): Promise<void> {
+  const currentTime = new Date().getTime();
```

Comment:
Just use Date.now(), returns the same value without the intermediate date object